pasta/augment/import_utils.py

```diff
@@ -53,3 +55,34 @@ def split_import(sc, node, alias_to_remove):
 
   parent_list.insert(idx + 1, new_import)
   return new_import
+
+def get_unused_imports(tree):
+  """Get the import nodes that aren't used.
+
+  Arguments:
+    tree: (ast.AST) An ast tree to find imports in.
+
+  Returns:
+    A list of ast.AST representing nodes that are imports or parts of imports
+    (aliases) that can be removed because they are unused.
+  """
+  sc = scope.analyze(tree)
+  unused_aliases = set()
+  for node in ast.walk(tree):
+    if isinstance(node, ast.alias):
+      name = sc.lookup_name(
+          node.asname if node.asname is not None else node.name)
+      if not name.reads:
+        unused_aliases.add(node)
+
+  unused_imports = []
+  for node in ast.walk(tree):
```

Comment:
Seems like you should be able to combine these loops. Just putt he body of your `isinstance(node, ast.alias)` check into the `for alias in node.names` block.

Reply:
Obsolete